A3-channel-tracing.md

```diff
@@ -21,48 +21,35 @@ This design uses the same terminology discussed in the [channelz](A14-channelz.m
 In addition, the following terminology will be used:
 
 1.  a "Trace event" is an interesting thing that happens to a channel. Examples include things like creation, address resolution, subchannel creation, connectivity state changes.
-2. "Trace data" is all of the data that is contained for one channel. This includes a list of _Trace events_, as well as metadata like the timestamp at which the channel was created.
-3.  a "Tracing object" is an in-memory data structure responsible for holding the _Trace data_ for a single channel.
+2.  a "Channel trace" is a data structure responsible for holding all data for a single channel. This includes a list of _Trace events_, as well as metadata like the timestamp at which the channel was created. The _Channel trace_ also holds a list of _Channel trace_ object of its child channels (if it has any).
 
 ## Abstract
 
-The document proposes adding a dedicated _Tracing object_ for every channel and subchannel. The _Tracing object_ will record important events in the life of a channel, like address resolution, subchannel creation, channel state changes, etc. The _Trace data_ from this _Tracing object_ will be made available through the [channelz service](A14-channelz.md).
+The document proposes adding a dedicated _Channel trace_ for every channel and subchannel. The _Channel trace_ will record important events in the life of a channel, like address resolution, subchannel creation, channel state changes, etc. The data from this _Channel trace_ will be made available through the [channelz service](A14-channelz.md).
 
 ## Background
 
 Channel connectivity issues are the root cause of a significant portion of user reported gRPC bugs. Channel tracing will be helpful for getting live channel data from a misbehaving program.
 
 ## Proposal
 
-To support channel tracing, implementations will use a dedicated _Tracing object_ that is owned by each channel and subchannel. We will add an API that allows library users to retrieve all of the _Tracing object's_ data for the channel in a language idiomatic way. Implementations must expose a toggle for this behavior (for example, C++ will add a new channel argument, GRPC_ARG_CHANNEL_TRACING).
+To support channel tracing, implementations will use a dedicated _Channel trace_ that is owned by each channel and subchannel. All of the data will be exposed via the [channelz service](A14-channelz.md). Implementations MAY add an API that allows library users to retrieve all of the _Tracing object's_ data for the channel in a language idiomatic way. Implementations must expose a toggle for this behavior (for example, C++ will add a new channel argument, GRPC_ARG_CHANNEL_TRACING).
```

Comment:
The wording of the last sentence needs to specify exactly what the knob does.  Currently, it just says "this behavior", but it's not clear exactly what that means.  I think the intent here is that this will inhibit collection of tracing data, but someone reading this would probably think it was referring to the exporting of the data, since that's what the rest of this paragraph talks about.

Also, the knob you've implemented in C-core is not just a toggle; it's actually a setting of how many trace events to keep per channel or subchannel, which disables collection if set to 0.  We should choose one of the following approaches:

1. Dictate the particular semantic of disabling collection when the max number of events is set to 0.  (In this case, we can probably move this to the end of the next paragraph.)

2. Require that implementations provide some API for disabling collection of trace events, but not dictate the exact semantics of how that API will work.  (In this case, the knob should probably be described in a paragraph of its own.)

Reply:
I have chosen option 1.